src/SqlAsyncEnumerable.cs

```diff
@@ -105,7 +105,10 @@ private async Task<bool> GetNextRowAsync()
                 {
                     using (SqlCommand command = SqlBindingUtilities.BuildCommand(this._attribute, this._connection))
                     {
-                        await command.Connection.OpenAsync();
+                        if (this._connection.State != System.Data.ConnectionState.Open)
+                        {
+                            await this._connection.OpenAsync();
```

Comment:
I still don't understand why this is needed. Why would the connection be closed between the constructor call and this?

Reply:
I debugged this and am not sure why but the MoveNextAsync is being invoked one more time after DisposeAsync() is called. So the connection is being closed in DisposeAsync and we're trying to load the reader again so the error. I've added connection check and to return false if the connection is closed.